packages/frontend-v2/components/Sidebar/Sidebar.tsx

```diff
@@ -54,6 +60,9 @@ const getRequiredCourses = (
   }
 };
 
+// A number to help avoid displaying stale validation info.
+let currentRequestNum = 0;
```

Comment:
Turns out that my previous strategy to avoid race conditions didn't work due to a quirk of web workers, so I moved the logic into the sidebar, which functions correctly. This did uncover the fact that the worker will do all previously requested validations before starting a new one, so with large plans, the user can end up waiting for a long time, unnecessarily. 

I think in a potential future ticket, I'm going to look into the performance implications of spawning a new worker each time, which would allow us to potentially cancel the useless stale validations entirely.

Reply:
I made a new issue that include some thoughts for future validation performance improvements (#535).

Reply:
ahh that's annoying, thanks for making the ticket and figuring this out!!